File: instrumentation/http4s-blaze-server-2.12_0.23/src/main/scala/com/nr/instrumentation/http4s/ResponseWrapper.scala
package com.nr.instrumentation.http4s

import com.newrelic.api.agent.{ExtendedResponse, HeaderType}
import org.http4s.{Header, Response}
import org.typelevel.ci.CIString

import scala.util.Try


class ResponseWrapper[F[_]](val response: Response[F]) extends ExtendedResponse {

  def getStatus: Int =
    response.status.code

  def getStatusMessage: String =
    response.status.reason

  def getContentType: String =
    response.headers.get(CIString("Content-Type"))
            .map(_.head.value)
            .orNull

  def getHeaderType: HeaderType =
    HeaderType.HTTP

  def setHeader(name: String, value: String): Unit = {
    response.headers.put(Header.Raw(CIString(name), value))
  }

  def getContentLength: Long =
    response.headers.get(CIString("Content-Length"))
            .flatMap(headers => Try(headers.head.value.toLong).toOption)
            .getOrElse(-1L)
}

object ResponseWrapper {
  def apply[F[_]](response: Response[F]): ExtendedResponse =
    new ResponseWrapper[F](response)
}

File: instrumentation/netty-4.0.37/src/main/java/com/agent/instrumentation/netty40/ResponseWrapper.java
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.agent.instrumentation.netty40;

import com.newrelic.api.agent.ExtendedResponse;
import com.newrelic.api.agent.HeaderType;
import io.netty.handler.codec.http.HttpHeaders;
import io.netty.handler.codec.http.HttpResponse;

public class ResponseWrapper extends ExtendedResponse {
    private final HttpResponse response;

    public ResponseWrapper(HttpResponse msg) {
        this.response = msg;
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public void setHeader(String name, String value) {
        response.headers().set(name, value);
    }

    @Override
    public int getStatus() throws Exception {
        return response.getStatus().code();
    }

    @Override
    public String getStatusMessage() throws Exception {
        return response.getStatus().reasonPhrase();
    }

    @Override
    public String getContentType() {
        return response.headers().get(HttpHeaders.Names.CONTENT_TYPE);
    }

    @Override
    public long getContentLength() {
        String contentLengthHeader = response.headers().get(HttpHeaders.Names.CONTENT_LENGTH);
        try {
            return contentLengthHeader != null ? Long.parseLong(contentLengthHeader) : -1;
        } catch (NumberFormatException e) {
            return -1;
        }
    }

}
